sdk/openai/openai-assistants/review/openai-assistants.api.md

```diff
@@ -0,0 +1,649 @@
+    // (undocumented)
+    retrieveThread: (threadId: string, options?: AssistantThreadsRetrieveThreadOptions) => Promise<AssistantThread>;
+}
+
+// @public (undocumented)
+export interface AssistantThreadsRetrieveThreadOptions extends OperationOptions {
+}
+
+// @public
+export interface CodeInterpeterCallDetails {
+    input: string;
+    outputs: CodeInterpreterCallOutput[];
+}
+
+// @public
+export interface CodeInterpreterCallOutput {
+    type: string;
+}
+
+// @public
+export interface CodeInterpreterImageReference {
+    fileId: string;
+}
+
+// @public
+export interface CreateAndRunThreadOptions {
+    assistantId: string;
+    instructions?: string;
+    metadata?: Record<string, string>;
+    model?: string;
+    thread?: AssistantThreadCreationOptions;
+    tools?: ToolDefinition[];
+}
+
+// @public
+interface File_2 {
+    bytes: number;
+    createdAt: Date;
+    filename: string;
+    id: string;
+    object: "file";
+    purpose: FilePurpose;
+}
+export { File_2 as File }
+
+// @public
+export interface FileDeletionStatus {
+    deleted: boolean;
+    id: string;
+    object: "file";
+}
+
+// @public
+export interface FileListResponse {
+    data: File_2[];
+    object: "list";
+}
+
+// @public
+export type FilePurpose = string;
+
+// @public (undocumented)
+export interface FilesDeleteFileOptions extends OperationOptions {
+}
+
+// @public (undocumented)
+export interface FilesListFilesOptions extends OperationOptions {
+    purpose?: FilePurpose;
+}
+
+// @public (undocumented)
+export interface FilesOperations {
+    // (undocumented)
+    deleteFile: (fileId: string, options?: FilesDeleteFileOptions) => Promise<FileDeletionStatus>;
+    // (undocumented)
+    listFiles: (options?: FilesListFilesOptions) => Promise<FileListResponse>;
+    // (undocumented)
+    retrieveFile: (fileId: string, options?: FilesRetrieveFileOptions) => Promise<File_2>;
+    // (undocumented)
+    retrieveFileContent: (fileId: string, options?: FilesRetrieveFileContentOptions) => Promise<Uint8Array>;
+    // (undocumented)
+    uploadFile: (file: Uint8Array, purpose: FilePurpose, options?: FilesUploadFileOptions) => Promise<File_2>;
+}
+
+// @public (undocumented)
+export interface FilesRetrieveFileContentOptions extends OperationOptions {
+}
+
+// @public (undocumented)
+export interface FilesRetrieveFileOptions extends OperationOptions {
+}
+
+// @public (undocumented)
+export interface FilesUploadFileOptions extends OperationOptions {
+    contentType?: string;
+    filename?: string;
+}
+
+// @public
+export interface FunctionCallDetails {
+    arguments: string;
+    name: string;
+    output?: string | null;
+}
+
+// @public
+export interface FunctionDefinition {
+    description: string;
+    name: string;
+    parameters: unknown;
+}
+
+// @public
+export interface ListResponseOf<T> {
+    data: T[];
+    firstId: string;
+    hasMore: boolean;
+    lastId: string;
+    object: "list";
```

Comment:
I think we can omit the `object` properties since they don't offer any benefits to SDK customers in the presence of types.